packages/react-native-reanimated/src/component/ReduceMotionConfig.tsx

```diff
@@ -0,0 +1,32 @@
+'use strict';
+import { useEffect, useRef } from 'react';
+import { ReduceMotion } from '../commonTypes';
+import { IS_REDUCED_MOTION } from '../hook/useReducedMotion';
+
+/**
+ * A component that lets you overwrite default reduce motion behavior globally in your application.
+ *
+ * @param mode - Determines default reduce motion behavior globally in your application. Configured with {@link ReduceMotion} enum.
+ * @see https://docs.swmansion.com/react-native-reanimated/docs/components/ReduceMotionConfig
+ */
+export function ReduceMotionConfig({ mode }: { mode: ReduceMotion }) {
+  const firstRender = useRef(true);
+  if (__DEV__ && firstRender.current) {
+    firstRender.current = false;
+    console.warn(
+      `[Reanimated] Reduced motion setting is overwritten with mode '${mode}'.`
+    );
+  }
```

Comment:
According to React purity dogma, we should move it to its own `useEffect` block with empty dependency array (runs just once on mount). Then, we don't need to read and write `firstRender` ref during render (which is against function purity).

Reply:
I think we even don't need to use refs if we put it into a `useEffect` with empty deps, since it then behaves like `componentDidMount` callback (and I think we want the warning every mount, don't we?)

Reply:
Update: this approach doesn't interfere with purity of functions BUT omitting using a flag is always a good idea